src/Metadata/PropertyTypeIterable.php

```diff
@@ -24,8 +34,10 @@ final class PropertyTypeIterable extends PropertyTypeArray
      */
     public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, string $collectionClass = null)
```

Comment:
does the collection class still need to be nullable, or can we require it?

Reply:
should we call it $traversableClass? the `@param` annotation already allows any traversable class

Reply:
we could still change the name in the 1.x branch too, i did not tag yet

Reply:
do you mean the change with using `\Traversable::class` as default? i think we should not do that but keep chosing instantiatable classes.

Reply:
glad if you can make it handle those cases, yes

Reply:
about the abstract classes: we could add configuration to the generator for concrete classes to use for interfaces abstract and seed that with the jms map you mention in your [other comment](https://github.com/liip/metadata-parser/pull/43#discussion_r1375506455) and then allow to add your own things (to handle custom traversable things that are required by a model)